crates/sui-core/src/authority.rs

```diff
@@ -523,6 +526,11 @@ impl AuthorityState {
         self.update_state(temporary_store, &certificate, &signed_effects)
             .await?;
 
+        // Each certificate only reaches here once
+        if let Some(event_handler) = &self.event_handler {
```

Comment:
I had another thought but we can address this in a separate PR.   I don't think we really need to have all the event processing completed as a part of the normal call to process a certificate, basically this adds latency to certificate processing, when clients don't really need event processing to be completed before the call returns.  So in the future we should move this out to be done in the side, like push to a stream or channel or something.

Reply:
certainly